packages/rum/test/viewCollection.spec.ts

```diff
@@ -99,6 +115,74 @@ describe('rum track renew session', () => {
   })
 })
 
+describe('rum track load duration', () => {
+  let lifeCycle: LifeCycle
+  let getViewEvent: (index: number) => View
+
+  beforeEach(() => {
+    jasmine.clock().install()
+    ;({ lifeCycle, getViewEvent } = spyOnViews())
+  })
+
+  afterEach(() => {
+    jasmine.clock().uninstall()
+  })
+
+  it('should collect intital view type as "initial_load"', () => {
+    expect(getViewEvent(0).loadType).toEqual(ViewLoadType.INITIAL_LOAD)
+  })
+
+  it('should collect view type as "route_change" after a route change', () => {
+    history.pushState({}, '', '/bar')
+    expect(getViewEvent(1).location.pathname).toEqual('/foo')
+    expect(getViewEvent(1).loadType).toEqual(ViewLoadType.INITIAL_LOAD)
+
+    expect(getViewEvent(2).location.pathname).toEqual('/bar')
+    expect(getViewEvent(2).loadType).toEqual(ViewLoadType.ROUTE_CHANGE)
+  })
+})
+
+describe('rum track load duration', () => {
+  let lifeCycle: LifeCycle
+  let getViewEvent: (index: number) => View
+
+  const mockDateNowT1 = 1500000000000
+  const mockDateNowT2 = 1500001000000
+
+  beforeEach(() => {
+    let isFirstPerformanceNow = true
+    spyOn(performance, 'now').and.callFake(() => {
+      if (isFirstPerformanceNow) {
+        isFirstPerformanceNow = false
+        return mockDateNowT1
+      }
+      return mockDateNowT2
+    })
+
+    jasmine.clock().install()
+    ;({ lifeCycle, getViewEvent } = spyOnViews())
+  })
+
+  afterEach(() => {
+    jasmine.clock().uninstall()
+  })
+
+  it('should set a loadingTime once the load is complete without having any activity', () => {
+    jasmine.clock().tick(AFTER_PAGE_ACTIVITY_MAX_DURATION)
+    jasmine.clock().tick(THROTTLE_VIEW_UPDATE_PERIOD)
+
+    expect(getViewEvent(1).loadingTime).toEqual(mockDateNowT2 - mockDateNowT1)
+  })
+
+  it('should set a loadingTime once the load is complete after having some activity', () => {
+    jasmine.clock().tick(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
+    lifeCycle.notify(LifeCycleEventType.DOM_MUTATED)
+    jasmine.clock().tick(AFTER_PAGE_ACTIVITY_END_DELAY)
+    jasmine.clock().tick(THROTTLE_VIEW_UPDATE_PERIOD)
+    expect(getViewEvent(1).loadingTime).toEqual(mockDateNowT2 - mockDateNowT1)
+  })
```

Comment:
For me, it is confusing that both tests have the same expectation:
```
.toEqual(mockDateNowT2 - mockDateNowT1)
```
Could the difference between the two behaviors be expressed more clearly in the expectation?

Reply:
Unfortunately the two scenarios uses the performance.now at different moments but they still need to be mocked as arbitrary values. Here we just want to make sure that both tests call performance.now twice and return the difference.